dashboard/app/views/levels/_dialogHelper.js

```diff
@@ -5,6 +5,48 @@
  * submit button interactions.
  */
 
+// Are we read-only?  This can be because we're a teacher OR because an answer
+// has been previously submitted.
+if (appOptions.readonlyWorkspace) {
+  // hide the Submit button.
+  $('.submitButton').hide();
+
+  // Are we a student viewing their own previously-submitted work?
+  if (appOptions.submitted) {
+    // show the Unsubmit button.
+    $('.unsubmitButton').show();
+  }
+
+  // Set the entire page background to be light grey.
+  $('.full_container').addClass('submitted_readonly');
+}
+
+// Unsubmit button should only be available when this is a standalone level.
+$('.unsubmitButton').click(function () {
```

Comment:
when building this i didn't use dialoghelper for the unsubmit button, but arguably should have.  now that we're moving this logic into dialoghelper itself, it might be correct to use its own showDialog instead?